portal-data/lib/index.js

```diff
@@ -2,197 +2,168 @@
 
 const Hoek = require('hoek');
 const Penseur = require('penseur');
-
+const DCClient = require('docker-compose-client');
+const awaitify = require('apr-awaitify');
 
 const internals = {
   defaults: {
     name: 'portal'
   }
 };
 
-
 module.exports = class Data {
-  constructor (options) {
+  constructor(options) {
     const settings = Hoek.applyToDefaults(options || {}, internals.defaults);
-    const name = settings.name;
-    delete settings.name;
 
     // Penseur will assert that the options are correct
-    this._db = new Penseur.Db(name, settings);
-  }
-
-  connect (cb) {
-    return new Promise((resolve, reject) => {
-      this._db.establish(['activities', 'datacenters', 'deployments', 'manifests', 'metrics'], (err) => {
-        if (typeof cb === 'function') {
-          return cb(err);
-        }
-
-        if (err) {
-          return reject(err);
-        }
-
-        resolve();
-      });
-    });
-  }
-
-  createDeployment (deployment) {
-    return new Promise((resolve, reject) => {
-      deployment.services = [];
-      deployment.state = { current: 'stopped' };
-
-      this._db.deployments.insert(deployment, (err, key) => {
-        if (err) {
-          return reject(err);
-        }
+    this._db = new Penseur.Db(settings.name, settings);
+    this._docker = new DCClient(settings.dockerHost);
+
+    this._db.establish = awaitify(this._db.establish);
+    this._db.deployments = awaitify(this._db.deployments);
+
+    // promisify Penseur
+    [
+      'activities',
+      'datacenters',
+      'deployments',
+      'manifests',
+      'metrics'
+    ].forEach(tableName =>
```

Comment:
that's fair, since you had only exposed promised methods I thought you were planning on only supporting promises ;) I'll update that  👍